### PR TITLE
Revert part of #10279

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

We shouldn't support this, as this is a bug in the DataLoader implementation.

Adding arbitrary `**kwargs` to a custom DataLoader allows for duplicate arguments.
This is something that should be checked by the user.

The fix that this PR removes only works for the first argument and if the name is "dataset", but this would still break for any of the other duplications.

Instead, the author of the DataLoader should write something like:

```python
def __init__(self, value: int = 2, *args, **kwargs):
    if "dataset" in kwargs:
        dataset = kwargs["dataset"]
    else:
        dataset = range(value)
    super().__init__(dataset, *args, **kwargs)
```

Partly reverts #10279
Part of https://github.com/PyTorchLightning/pytorch-lightning/issues/10329

A follow-up PR will add a `MisconfigurationException` to help the user in these cases.

### Does your PR introduce any breaking changes? If yes, please list them.

This will no longer work for anybody already relying on this. Lite is experimental

## Before submitting

- [ ] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: tests/lite/test_lite.py [hunk 1]
```diff
@@ -164,57 +164,6 @@ def test_setup_dataloaders_return_type():
     assert lite_dataloader1.dataset is dataset1
 
 
-def test_setup_custom_dataloaders():
```
Comment (tchaton) [inline tests/lite/test_lite.py:167]: We could have a custom dataloader using only a dataset as a test.

**Review by tchaton — APPROVED** [review]
LGTM !

File: tests/lite/test_lite.py (hunk 1 above)
Comment (carmocca) [reply tests/lite/test_lite.py:167]: We already test that in `tests/trainer/test_data_loading.py`

**Review by awaelchli — APPROVED** [review]

File: tests/lite/test_lite.py (hunk 1 above)
Comment (tchaton) [reply tests/lite/test_lite.py:195]: I meant, we should unittest the context manager to make sure the params are

**Review by tchaton — APPROVED** [review]
LGTM ! @carmocca I added a test to ensure the new context manager is being tested.

**Review by rohitgr7 — APPROVED** [review]

File: tests/lite/test_lite.py (hunk 1 above)
Comment (carmocca) [reply tests/lite/test_lite.py:195]: @tchaton isn't that covered by the test that @awaelchli added in https://github.com/PyTorchLightning/pytorch-lightning/pull/10334?
